### PR TITLE
feat: better error report for ambiguous export

### LINKED ISSUE #1643: Better error report for ambiguous export
https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.js%00import+%7B+foo+%7D+from+%27.%2Fproxy%27%0Aconsole.log%28foo%29&b=%00a.js%00export+const+foo+%3D+1&b=%00b.js%00export+const+foo+%3D+2&b=%00proxy.js%00export+*+from+%27.%2Fa%27%0Aexport+*+from+%27.%2Fb%27&o=--bundle+--splitting+--format%3Desm+--external%3Aext+--outdir%3Ddist

Esbuild treats this as an error and has better report. Let's merge this PR for now. I'll open an new issue to port esbuild's behavior.

_Originally posted by @hyf0 in https://github.com/rolldown/rolldown/pull/1634#discussion_r1679021003_

A follow up of #1621.

### PR DESCRIPTION
**Pull request by shulaoda**
### Description

Related #1643

### CODE REVIEW

**Comment by shulaoda** [comment]
I am unable to continue working because the`named_exports` property of `EcmaModule` is insufficient to support my work. I have attempted to modify it, and the following is the draft result: 

https://github.com/rolldown/rolldown/blob/d78153f536c4dc132ede4c0d1d987e0cd5234efb/crates/rolldown/tests/fixtures/errors/ambiguous_external_namespace/artifacts.snap#L11-L35

@hyf0

File: crates/rolldown_error/src/events/ambiguous_external_namespace.rs [hunk 1]
```diff
@@ -4,14 +4,19 @@ use oxc::span::Span;
...
+
 #[derive(Debug)]
 pub struct AmbiguousExternalNamespace {
-  pub importer: String,
-  pub importee: Vec<String>,
-  pub importer_source: ArcStr,
-  pub importer_filename: String,
-  pub imported_specifier: String,
-  pub imported_specifier_span: Span,
+  pub symbol: String,
```
Comment (hyf0) [inline crates/rolldown_error/src/events/ambiguous_external_namespace.rs]:
```suggestion
  pub ambiguous_export_name: String,
```

File: crates/rolldown_error/src/events/ambiguous_external_namespace.rs [hunk 2]
```diff
@@ -4,14 +4,19 @@ use oxc::span::Span;
 
 use crate::{diagnostic::Diagnostic, types::diagnostic_options::DiagnosticOptions, EventKind};
 
+#[derive(Debug)]
+pub struct Namespace {
```
Comment (hyf0) [inline crates/rolldown_error/src/events/ambiguous_external_namespace.rs]: The naming here is quite confused. I can't think of a better one now. But let's not use `Namespace` such a common name.

```suggestion
pub struct AmbiguousExternalNamespaceModule {
```

File: crates/rolldown_error/src/events/ambiguous_external_namespace.rs [hunk 3]
```diff
@@ -4,14 +4,19 @@ use oxc::span::Span;
...
+  pub span: Span,
```
Comment (hyf0) [inline crates/rolldown_error/src/events/ambiguous_external_namespace.rs]:
```suggestion
  // Point to `import { [identifier] } from ...` or `export [identifier]`
  pub span_of_ identifier: Span,
```

**Comment by hyf0** [comment]
Changes look fine except those commented.

CI failed due to lint and tests. Please read 
- https://rolldown.rs/contrib-guide/testing
- https://rolldown.rs/contrib-guide/building-and-running

to fix them.

**Comment by shulaoda** [comment]
> Changes look fine except those commented.
> 
> CI failed due to lint and tests. Please read
> 
> * https://rolldown.rs/contrib-guide/testing
> * https://rolldown.rs/contrib-guide/building-and-running
> 
> to fix them.

I don't know if I should change this fixture `filename_with_hash` snapshot.

**Comment by hyf0** [comment]
> > Changes look fine except those commented.
> > CI failed due to lint and tests. Please read
> > 
> > * https://rolldown.rs/contrib-guide/testing
> > * https://rolldown.rs/contrib-guide/building-and-running
> > 
> > to fix them.
> 
> I don't know if I should change this fixture `filename_with_hash` snapshot.

Yeah. You should change it.

**Review by hyf0 — APPROVED** [review]
Well done!

**Pull request merged**